executor/hnswpsql.py

```diff
@@ -330,3 +336,23 @@ def cleanup(self, **kwargs):
             self._kv_indexer.cleanup()
         else:
             self.logger.warning(f'PSQL has not been initialized')
+
+    def _start_auto_sync(self):
+        self.sync_thread = Thread(target=self._sync_loop, daemon=True)
+        self.sync_thread.start()
+
+    def close(self) -> None:
+        if hasattr(self, 'sync_thread'):
+            try:
+                self.sync_thread.join(1)
```

Comment:
u should not join, u should use some threading Event to communicate that it needs to terminate